src/Toolkit/Html.php

```diff
@@ -446,128 +447,110 @@ public static function video(string $url, array $options = [], array $attr = [])
-            ],
-
-            // https://www.youtube.com/embed/d9NF2edxy-M
-            // https://www.youtube.com/embed/d9NF2edxy-M?start=10
-            ['pattern' => 'youtube.com\/embed\/([a-zA-Z0-9_-]+)(?:\?(.+))?'],
-
-            // https://www.youtube-nocookie.com/embed/d9NF2edxy-M
-            // https://www.youtube-nocookie.com/embed/d9NF2edxy-M?start=10
-            [
-                'pattern' => 'youtube-nocookie.com\/embed\/([a-zA-Z0-9_-]+)(?:\?(.+))?',
-                'domain'  => 'www.youtube-nocookie.com'
-            ],
-
-            // https://www.youtube-nocookie.com/watch?v=d9NF2edxy-M
-            // https://www.youtube-nocookie.com/watch?v=d9NF2edxy-M&t=10
-            [
-                'pattern' => 'youtube-nocookie.com\/watch\?v=([a-zA-Z0-9_-]+)(?:&(.+))?',
-                'domain'  => 'www.youtube-nocookie.com'
-            ],
-
-            // https://www.youtube-nocookie.com/playlist?list=PLj8e95eaxiB9goOAvINIy4Vt3mlWQJxys
-            [
-                'pattern' => 'youtube-nocookie.com\/playlist\?list=([a-zA-Z0-9_-]+)',
-                'domain'  => 'www.youtube-nocookie.com',
-                'uri'     => 'embed/videoseries?list='
-            ],
-
-            // https://www.youtube.com/watch?v=d9NF2edxy-M
-            // https://www.youtube.com/watch?v=d9NF2edxy-M&t=10
-            ['pattern' => 'youtube.com\/watch\?v=([a-zA-Z0-9_-]+)(?:&(.+))?'],
-
-            // https://www.youtube.com/playlist?list=PLj8e95eaxiB9goOAvINIy4Vt3mlWQJxys
-            [
-                'pattern' => 'youtube.com\/playlist\?list=([a-zA-Z0-9_-]+)',
-                'uri'     => 'embed/videoseries?list='
-            ],
-
-            // https://youtu.be/d9NF2edxy-M
-            // https://youtu.be/d9NF2edxy-M?t=10
-            ['pattern' => 'youtu.be\/([a-zA-Z0-9_-]+)(?:\?(.+))?']
-        ];
-
-        foreach ($schemes as $schema) {
-            if (preg_match('!' . $schema['pattern'] . '!i', $url, $array) === 1) {
-                $domain = $schema['domain'] ?? $domain;
-                $uri    = $schema['uri'] ?? $uri;
-                $id     = $array[1];
-                if (isset($array[2]) === true) {
-                    parse_str($array[2], $urlOptions);
-
-                    // convert video URL options to embed URL options
-                    if (isset($urlOptions['t']) === true) {
-                        $urlOptions['start'] = $urlOptions['t'];
-                        unset($urlOptions['t']);
-                    }
+        if (preg_match('!youtu!i', $url) !== 1) {
+            throw new Exception('Invalid YouTube source');
+        }
+
+        $uri    = new Uri($url);
+        $path   = $uri->path();
+        $query  = $uri->query();
+        $first  = $path->first();
+        $second = $path->nth(1);
+        $host   = 'https://' . $uri->host() . '/embed';
+        $src    = null;
+
+        $isYoutubeId = function (string $id): bool {
+            return preg_match('!^[a-zA-Z0-9_-]+$!', $id);
```

Comment:
`preg_match()` does not return a bool, but `1|0|false`. So Psalm will complain about this on higher levels. Fix:

```suggestion
            return preg_match('!^[a-zA-Z0-9_-]+$!', $id) === 1;
```